src/scripts/autosplitter.gd

```diff
@@ -3,54 +3,45 @@ extends Node
 
 signal timer_updated
 
-# TODO: Godot 4 remove WebSocketServer, and creating a WebSocket server is different now and not as documented
-#var server: WebSocketServer
-#var peers := []
+var server := TCPServer.new()
+var peers: Array[WebSocketPeer] = []
 
 var speedrun_is_running := false
 var run_started_time: int
 var speedrun_time: float
 
 
 func _ready() -> void:
-	#server = WebSocketServer.new()
-	#server.client_close_request.connect(self._on_server_client_close_request)
-	#server client_connected.connect(self._on_server_client_connected)
-	#server.client_disconnected.connect(self._on_server_client_disconnected)
-	#server.data_received.connect(self._on_server_data_received)
 	process_mode = Node.PROCESS_MODE_ALWAYS
 
 
 func start() -> void:
-	pass
-	#if server.is_listening():
-	#	return
-	#var error := server.listen(Global.autosplitter_port)
-	#if error:
-	#	print("Error starting websocket, autosplitter will not run")
-	#	return
-	#Global.autosplitter_enabled = true
-	#print("Websocket listening on %s:%d" % [server.bind_ip, Global.autosplitter_port])
+	if server.is_listening():
+		return
+	var error := server.listen(Global.autosplitter_port)
+	if error:
+		push_error("Error starting websocket server, autosplitter will not run!")
```

Comment:
Do we have some sort of warning to the user outside of the console? Like a popup or something. If we don't might be good to add.

Reply:
The player will know it didn't work because the button will uncheck itself if the autosplitter didn't activate successfully: https://github.com/kdreese/gmtk-2022/blob/6cb4a438e0ffb4d92efd785b7fe13bef00633fb4/src/states/options_menu.gd#LL94C1-L94C1

There is no explicit warning though, and I think it'd be a good idea to add one. I'll add a small bit of text in the options menu or something